### PR TITLE
Memory updated

### PR DESCRIPTION
**Pull request by javipacheco**
This PR closes issue #466

### CODE REVIEW

File: core/src/commonMain/kotlin/com/xebia/functional/xef/llm/MemoryManagement.kt [hunk 1]
```diff
@@ -4,85 +4,51 @@ import com.xebia.functional.xef.conversation.Conversation
...
-  )
+internal fun Message.toMemory(cid: ConversationId, index: Int): Memory =
+  Memory(conversationId = cid, content = this, index = index)
 
-internal fun List<Message>.toMemory(chat: LLM, scope: Conversation): List<Memory> {
+internal fun List<Message>.toMemory(scope: Conversation): List<Memory> {
   val cid = scope.conversationId
   return if (cid != null) {
-    mapIndexed { index, it -> it.toMemory(cid, chat, index) }
+    mapIndexed { index, it -> it.toMemory(cid, scope.store.incrementIndexAndGet()) }
```
Comment (Montagon) [inline core/src/commonMain/kotlin/com/xebia/functional/xef/llm/MemoryManagement.kt:26]: I think we are not using the index, right? Could we change it to `map`?

**Review by Montagon — APPROVED** [review]
Great job @javipacheco 👏

File: core/src/commonMain/kotlin/com/xebia/functional/xef/llm/MemoryManagement.kt (hunk 1 above)
Comment (javipacheco) [reply core/src/commonMain/kotlin/com/xebia/functional/xef/llm/MemoryManagement.kt:26]: Good catch

**Review by raulraja — DISMISSED** [review]
👍

**Review by Montagon — APPROVED** [review]

**Pull request merged**